src/db_tables/user_list.py

```diff
@@ -0,0 +1,96 @@
+from enum import Enum
+from typing import ClassVar, TYPE_CHECKING
+
+from sqlalchemy import ForeignKey, Index, UniqueConstraint
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from src.utils.database import Base
+
+if TYPE_CHECKING:
+    from src.db_tables.media import Episode, Movie, Series
+    from src.db_tables.user import User
+
+
+class ItemKind(Enum):
+    """Enum to represent the kind of item in a user list."""
+
+    SERIES = "series"
+    MOVIE = "movie"
+    EPISODE = "episode"
+    MEDIA = "media"  # either series or movie
+    ANY = "any"
+
+
+class UserList(Base):
+    """Table to store user lists."""
```

Comment:
```suggestion
    """Table to store user lists.
    
    This provides a dynamic way to store various lists of media for the user, such as favorites, to watch,
    already watched, ... all  tracked in the same table, instead of having to define tables for each such
    structure.
    """
```